docs/tutorial/code/login.py

```diff
@@ -0,0 +1,53 @@
+from jinja2 import Environment, PackageLoader
+from klein import Klein
+
+
+class MyApp(object):
+    app = Klein()
+
+    def __init__(self, db_name):
+        loader = PackageLoader("templates", package_path="")
+        templates = Environment(loader=loader)
+        self.templates = templates
+        self.db_name = db_name
+        self.sessions = {}
+        self.users = {
+            "frank": "hunter2",
+            "marry": "strong_password",
+            "amanda": "twisted12"
+        }
+
+    @app.route("/login")
+    def login_page(self, equest):
+        template = self.templates.get_template("login.html")
+        return template.render()
+
+    @app.route("/do_login", methods=["POST"])
```

Comment:
This form processing ignores the new form work I've been working towards in [this branch](https://github.com/glyph/klein/tree/form-hacks) and, as a result, is insecure by default; it is vulnerable to [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery) attacks.

Fixing this is tricky, which is exactly why that branch has been outstanding for some time.  Would you be interested in helping with it? :)